### PR TITLE
[Mellanox]platform api support firmware install

### PR DESCRIPTION
**Pull request by stephenxs**
**- What I did**
support firmware install, including CPLD and BIOS.

**- How I did it**
CPLD: cpldupdate
BIOS: boot to onie and update BIOS in onie and then boot to SONiC

**- How to verify it**
try installing CPLD/BIOS via platform api
[test-install-sonic-then-update-bios.txt](https://github.com/Azure/sonic-buildimage/files/3982693/test-install-sonic-then-update-bios.txt)
[test-update-cpld-gpio.txt](https://github.com/Azure/sonic-buildimage/files/3982695/test-update-cpld-gpio.txt)

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by jleveque** [comment]
Retest vs please
Retest vsimage please

**Comment by jleveque** [comment]
Retest vs please

**Comment by jleveque** [comment]
Retest vsimage please

**Comment by jleveque** [comment]
Retest vsimage please

**Comment by liat-grozovik** [comment]
@srideepDell could you please approve this PR or you have additional feedback?

**Comment by nazariig** [comment]
@stephenxs we need to improve our flows one https://github.com/Azure/sonic-buildimage/pull/3935 is merged

**Comment by stephenxs** [comment]
@nazariig the code that take advantage of new grub variable to resolve the sonic_installer broken issue has been merged.

File: platform/mellanox/onie-fw-update [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+}
+
+clean_onie_access()
+{
+	rm -f /lib/onie
+	umount $onie_partition
+}
+
+# ONIE entry must exist in grub config
+find_onie_menuentry()
```
Comment (nazariig) [inline platform/mellanox/onie-fw-update]: @stephenxs Since we are already use `onie_entry`, we do not need this anymore

Comment (stephenxs) [reply platform/mellanox/onie-fw-update]: removed.

**Comment by nazariig** [comment]
retest vsimage please

**Pull request merged**